### DISCUSSION TITLE
Upgrade to 2.0 int to bigint

### DISCUSSION THREAD

**Discussion by develocode777** [serverpod/serverpod#2394] [Q&A] [answered]
Hello,

Is it necessary to migrate all tables to bigint, including those used internally by serverpod, or just those created by the user?

Thank you.

**Comment by SandPod** [comment, marked as answer]
Hey!

We have written some guidance around this in the migration guide: https://docs.serverpod.dev/upgrading/upgrade-to-two#integer-representation-changed-to-bigint

As stated there:

> No manual modification to the database is required if this data representation is not essential for the application. However, all new migrations will be created with the new representation.

If the big int representation is not crucial for your application, no action is needed.
If it is desired, you are responsible for migrating all tables.

If you have not yet shipped your server the easiest way to migrate is to remove all existing migrations create a new one and  recreate your database (you can also migrated all your data from the database using [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) if needed). If the application is already shipped I would recommend checking out the resources linked in the documentation.

**Reply by develocode777** [reply]
Thanks!
